[2023-12-19T19:42:19.674+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: update_dwh.sat_employee_working_performance_update manual__2023-12-19T19:42:07.998635+00:00 [queued]>
[2023-12-19T19:42:19.685+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: update_dwh.sat_employee_working_performance_update manual__2023-12-19T19:42:07.998635+00:00 [queued]>
[2023-12-19T19:42:19.685+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2023-12-19T19:42:19.698+0000] {taskinstance.py:1382} INFO - Executing <Task(PostgresOperator): sat_employee_working_performance_update> on 2023-12-19 19:42:07.998635+00:00
[2023-12-19T19:42:19.711+0000] {standard_task_runner.py:57} INFO - Started process 1138 to run task
[2023-12-19T19:42:19.715+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'update_dwh', 'sat_employee_working_performance_update', 'manual__2023-12-19T19:42:07.998635+00:00', '--job-id', '385', '--raw', '--subdir', 'DAGS_FOLDER/update_dwh_dag.py', '--cfg-path', '/tmp/tmplzo7virk']
[2023-12-19T19:42:19.722+0000] {standard_task_runner.py:85} INFO - Job 385: Subtask sat_employee_working_performance_update
[2023-12-19T19:42:19.779+0000] {task_command.py:416} INFO - Running <TaskInstance: update_dwh.sat_employee_working_performance_update manual__2023-12-19T19:42:07.998635+00:00 [running]> on host 5e0c64ad45d7
[2023-12-19T19:42:19.913+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='update_dwh' AIRFLOW_CTX_TASK_ID='sat_employee_working_performance_update' AIRFLOW_CTX_EXECUTION_DATE='2023-12-19T19:42:07.998635+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2023-12-19T19:42:07.998635+00:00'
[2023-12-19T19:42:19.915+0000] {sql.py:274} INFO - Executing: 
INSERT INTO rdv.sat_employee_working_performance
SELECT DISTINCT
	md5(so.employeenumber::text),
	'csv',
	now() ,
	so.performancerating,
	so.businesstravel,
	so.jobinvolvement,
	case when so.overtime = 'Yes' then true else false end,
	so.trainingtimeslastyear
FROM
    staging.stg_table so
LEFT OUTER JOIN rdv.sat_employee_working_performance sat ON (sat.employee_hash_key = md5(so.employeenumber::text) AND sat.created_at IS NULL)
where 
	coalesce(so.performancerating, 0) != coalesce(sat.performance_rating, 0) or
	coalesce(so.businesstravel, '') != coalesce(sat.business_travel, '') or
	coalesce(so.jobinvolvement, 0) != coalesce(sat.job_involvement, 0) or
	coalesce(case when so.overtime = 'Yes' then true else false end, false) != coalesce(sat.overtime, false) or
	coalesce(so.trainingtimeslastyear, 0) != coalesce(sat.training_times_last_year, 0)
[2023-12-19T19:42:19.932+0000] {base.py:73} INFO - Using connection ID 'stg_conn' for task execution.
[2023-12-19T19:42:19.938+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/models/connection.py:374: AirflowProviderDeprecationWarning: The "schema" arg has been renamed to "database" as it contained the database name.Please use "database" to set the database name.
  return hook_class(**{hook.connection_id_attribute_name: self.conn_id}, **hook_params)

[2023-12-19T19:42:19.960+0000] {base.py:73} INFO - Using connection ID 'stg_conn' for task execution.
[2023-12-19T19:42:20.036+0000] {sql.py:418} INFO - Running statement: 
INSERT INTO rdv.sat_employee_working_performance
SELECT DISTINCT
	md5(so.employeenumber::text),
	'csv',
	now() ,
	so.performancerating,
	so.businesstravel,
	so.jobinvolvement,
	case when so.overtime = 'Yes' then true else false end,
	so.trainingtimeslastyear
FROM
    staging.stg_table so
LEFT OUTER JOIN rdv.sat_employee_working_performance sat ON (sat.employee_hash_key = md5(so.employeenumber::text) AND sat.created_at IS NULL)
where 
	coalesce(so.performancerating, 0) != coalesce(sat.performance_rating, 0) or
	coalesce(so.businesstravel, '') != coalesce(sat.business_travel, '') or
	coalesce(so.jobinvolvement, 0) != coalesce(sat.job_involvement, 0) or
	coalesce(case when so.overtime = 'Yes' then true else false end, false) != coalesce(sat.overtime, false) or
	coalesce(so.trainingtimeslastyear, 0) != coalesce(sat.training_times_last_year, 0), parameters: None
[2023-12-19T19:42:20.075+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/operators/sql.py", line 280, in execute
    output = hook.run(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 392, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 423, in _run_command
    cur.execute(sql_statement)
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "sat_employee_working_performance_pk"
DETAIL:  Key (employee_hash_key, created_at)=(3a066bda8c96b9478bb0512f0a43028c, 2023-12-19 19:42:20.02401+00) already exists.

[2023-12-19T19:42:20.101+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=update_dwh, task_id=sat_employee_working_performance_update, execution_date=20231219T194207, start_date=20231219T194219, end_date=20231219T194220
[2023-12-19T19:42:20.118+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 385 for task sat_employee_working_performance_update (duplicate key value violates unique constraint "sat_employee_working_performance_pk"
DETAIL:  Key (employee_hash_key, created_at)=(3a066bda8c96b9478bb0512f0a43028c, 2023-12-19 19:42:20.02401+00) already exists.
; 1138)
[2023-12-19T19:42:20.148+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2023-12-19T19:42:20.171+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
